File: rxpreferences/src/test/java/dev/drewhamilton/rxpreferences/PreferenceType.java
package dev.drewhamilton.rxpreferences;

import java.util.Set;

enum PreferenceType {
    STRING(String.class),
    STRING_SET(Set.class),
    INT(Integer.class),
    LONG(Long.class),
    FLOAT(Float.class),
    BOOLEAN(Boolean.class);

    final Class<?> valueClass;

    PreferenceType(Class<?> valueClass) {
        this.valueClass = valueClass;
    }
}

File: ktx/src/test/java/drewhamilton/rxpreferences3/PreferenceType.kt
package drewhamilton.rxpreferences3

enum class PreferenceType(val valueClass: Class<*>) {
    STRING(String::class.java),
    STRING_SET(Set::class.java),
    INT(Int::class.java),
    LONG(Long::class.java),
    FLOAT(Float::class.java),
    BOOLEAN(Boolean::class.java)
}
